src/main/webapp/2Dscript.js

```diff
@@ -80,7 +151,279 @@ function addCenterControl(controlDiv) {
   });
 }
+
+function createSideTable(json) {
+  const table = document.createElement('table');
+  let headerRow = document.createElement('tr');
+  let columnOne = document.createElement("th");
+  columnOne.innerText = "RunID";
+  let columnTwo = document.createElement("th");
+  columnTwo.innerText = "Select";
+  headerRow.appendChild(columnOne);
+  headerRow.appendChild(columnTwo);
+  table.appendChild(headerRow);
+  let even = false;
+  for (const key in json) {
+    let currentRow = document.createElement('tr');
+    currentRow.className = "visible";
+    if (even) {
+      currentRow.className += " even";
+    }
+    let keyEntry = document.createElement('td');
+    keyEntry.innerText = key;
+    currentRow.appendChild(keyEntry);
+
+    let checkBoxEntry = document.createElement("td");
+    const input = document.createElement("input");
+    input.type = "checkbox";
+    input.id = key;
+    input.addEventListener("click", function (event) {
+      // Get the necessary content for this runId if not stored in the local cache.
+      console.log("Checkbox event");
+      const checkboxElement = event.target;
+      console.log("The id is:" + event.target.id);
+      console.log("The event is" + event.target.toString());
+      console.log("The checkbox is " + this.checked);
+      if (this.checked) {
+        // First check the cache for this value.
+        dataEntries = datas[event.target.id];
+        console.log("The value of dataentries is " + dataEntries);
+        // if the value is not found in the cache then fetch it.
+        if (dataEntries === undefined) {
+          console.log("Data Entries was false");
+          console.log("The id is:" + event.target.id);
+          fetch("/getrun?id=" + event.target.id + "&dataType=pose").then(response => response.json())
+            .then(data => dataEntries = data).then(() => {
+              datas[event.target.id] = dataEntries;
+              const toGraph = plotLine(dataEntries);
+              maps[event.target.id] = toGraph;
+              toGraph.setMap(map);
+            });
+        }
+        else {
+          // Still plot the line, doesn't need to be asynchronous.
+          // TODO(morleyd): abstract this out into a function.
+          console.log("Dataentries was true");
+          const toGraph = plotLine(dataEntries);
+          maps[event.target.id] = toGraph
+          toGraph.setMap(map);
+        }
+      }
+      else {
+        // In this case the box has become unchecked. We want to remove this graph.
+        // We can only remove this box if it has been generated before and then can be removed appropriately.
+        const toRemove = maps[event.target.id];
+        if (toRemove)
+          toRemove.setMap(null);
+      }
+
+    });
+    checkBoxEntry.appendChild(input);
+    currentRow.appendChild(checkBoxEntry);
+    table.appendChild(currentRow);
+    console.log(currentRow);
+    even = !even;
+
+  }
```

Comment:
This loop is huge; please refactor into smaller functions.

Reply:
Yeah was working on that wanted to get it pushed though, so people could see what was going on.